contracts/contracts/VerificationGateway.sol

```diff
@@ -162,13 +182,21 @@ contract VerificationGateway
         // ensure first parameter is the calling wallet address
         bytes memory encodedAddress = abi.encode(address(wallet));
         uint8 selectorOffset = 4;
+        bytes4 selectorId = 0;
         for (uint256 i=0; i<32; i++) {
+            if (i<4) {
+                selectorId |= bytes4(encodedFunction[i]) >> i*8;
+            }
```

Comment:
Are you sure this is the correct shift direction? Usually you'd shift left in this situation (to move a byte into the more significant location, since `<<`, `>>` follows big endian convention, even though architectures are usually little endian).

I can see how it might be inverted for `bytes4`... curious.

It might be better to use `uint32` for the arithmetic and then coerce to `bytes4` after?